tests/test_modeling_utils.py

```diff
@@ -787,6 +792,27 @@ def test_legacy_load_from_url(self):
             "https://huggingface.co/hf-internal-testing/tiny-random-bert/resolve/main/pytorch_model.bin", config=config
         )
 
+    @require_torch_non_multi_gpu
+    def test_save_offloaded_model(self):
+        model_id = "bigscience/bloomz-560m"
```

Comment:
It would be great to use a smaller model with a custom `device_map`. This way, we won't have to worry about the `infer_auto_device_map` function.  Copied from `test_from_pretrained_disk_offload_task_model`
```py
model = AutoModel.from_pretrained("hf-internal-testing/tiny-random-gpt2").
device_map = {
    "transformer.wte": 0,
    "transformer.wpe": 0,
    "transformer.h.0": "cpu",
    "transformer.h.1": "cpu",
    "transformer.h.2": "cpu",
    "transformer.h.3": "disk",
    "transformer.h.4": "disk",
    "transformer.ln_f": 0,
    "lm_head": 0,
}
```

Reply:
Agreed, and the test has been modified to use this model and device configuration